AsyncEnumerator.cs

```diff
@@ -12,6 +12,7 @@ public sealed class AsyncEnumerator<T> : IAsyncEnumerator<T>
         /// This exception is thrown when you call <see cref="Yield.Break"/>
         /// </summary>
         public sealed class AsyncEnumerationCanceledException : OperationCanceledException { }
+        public sealed class AsyncEnumerationDisposedException : OperationCanceledException { }
```

Comment:
I think it make sense to derive this from `AsyncEnumerationCanceledException`, because an enumeration is canceled because the enumerator is disposed.